core/src/main/java/org/apache/brooklyn/core/location/dynamic/DynamicLocation.java

```diff
@@ -47,4 +49,14 @@
 
     E getOwner();
 
+    /**
+     * An opportunity to register this location (e.g. with the {@link LocationRegistry} or the 
+     * catalog, so that it will be persisted).
+     */
+    LocationDefinition register();
```

Comment:
`register()` is being used (in Clocker) to register the new "BasicLocationDefinition" for that new named location. We could maybe do it in init(), rebind() and unmanage(). But by having an explicit register() method called by the entity that owns the location, we can ensure that the LocationRegistry is only updated when the location object is completely initialised.